File: app/schemas/parent_ai.py
"""Schemas for Responsible AI Parent Tools."""
from pydantic import BaseModel, Field
from typing import Optional


# ── Requests ──

class WeakSpotsRequest(BaseModel):
    student_id: int
    course_id: int | None = None


class ReadinessCheckRequest(BaseModel):
    student_id: int
    assignment_id: int


class PracticeProblemsRequest(BaseModel):
    student_id: int
    course_id: int
    topic: str = Field(min_length=1, max_length=500)


# ── Responses ──

class WeakSpot(BaseModel):
    topic: str
    severity: str  # "high", "medium", "low"
    detail: str
    quiz_score_summary: str | None = None
    suggested_action: str


class WeakSpotsResponse(BaseModel):
    student_name: str
    course_name: str | None = None
    weak_spots: list[WeakSpot]
    summary: str
    total_quizzes_analyzed: int = 0
    total_assignments_analyzed: int = 0


class ReadinessItem(BaseModel):
    label: str
    status: str  # "done", "partial", "missing"
    detail: str | None = None


class ReadinessCheckResponse(BaseModel):
    student_name: str
    assignment_title: str
    course_name: str
    readiness_score: int = Field(ge=1, le=5)
    summary: str
    items: list[ReadinessItem]


class PracticeProblem(BaseModel):
    number: int
    question: str
    hint: str | None = None


class PracticeProblemsResponse(BaseModel):
    student_name: str
    course_name: str
    topic: str
    problems: list[PracticeProblem]
    instructions: str

File: frontend/src/api/parentAI.ts
import { api, AI_TIMEOUT } from './client';

export interface WeakSpot {
  topic: string;
  severity: 'high' | 'medium' | 'low';
  detail: string;
  quiz_score_summary: string | null;
  suggested_action: string;
}

export interface WeakSpotsResponse {
  student_name: string;
  course_name: string | null;
  weak_spots: WeakSpot[];
  summary: string;
  total_quizzes_analyzed: number;
  total_assignments_analyzed: number;
}

export interface ReadinessItem {
  label: string;
  status: 'done' | 'partial' | 'missing';
  detail: string | null;
}

export interface ReadinessCheckResponse {
  student_name: string;
  assignment_title: string;
  course_name: string;
  readiness_score: number;
  summary: string;
  items: ReadinessItem[];
}

export interface PracticeProblem {
  number: number;
  question: string;
  hint: string | null;
}

export interface PracticeProblemsResponse {
  student_name: string;
  course_name: string;
  topic: string;
  problems: PracticeProblem[];
  instructions: string;
}

export const parentAIApi = {
  getWeakSpots: (studentId: number, courseId?: number) =>
    api.post<WeakSpotsResponse>('/api/parent-ai/weak-spots', {
      student_id: studentId,
      course_id: courseId ?? null,
    }, AI_TIMEOUT),

  checkReadiness: (studentId: number, assignmentId: number) =>
    api.post<ReadinessCheckResponse>('/api/parent-ai/readiness-check', {
      student_id: studentId,
      assignment_id: assignmentId,
    }, AI_TIMEOUT),

  generatePracticeProblems: (studentId: number, courseId: number, topic: string) =>
    api.post<PracticeProblemsResponse>('/api/parent-ai/practice-problems', {
      student_id: studentId,
      course_id: courseId,
      topic,
    }, AI_TIMEOUT),
};
